sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs

```diff
@@ -199,16 +241,30 @@ internal static RecognizeCustomEntitiesResultCollection ConvertToRecognizeCustom
         {
             var recognizeEntities = new List<RecognizeEntitiesResult>(results.Errors.Count);
 
-            //Read errors
+            // Read errors.
             foreach (DocumentError error in results.Errors)
             {
                 recognizeEntities.Add(new RecognizeEntitiesResult(error.Id, ConvertToError(error.Error)));
             }
 
-            //Read document entities
-            foreach (var docEntities in results.Documents)
+            // Read document entities.
+            foreach (var document in results.Documents)
             {
-                recognizeEntities.Add(new RecognizeEntitiesResult(docEntities.Id, docEntities.Statistics ?? default, ConvertToCategorizedEntityCollection(docEntities)));
+                DetectedLanguage? detectedLanguage = (document.DetectedLanguage is not null)
+                    ? new DetectedLanguage(
+                        document.DetectedLanguage.Value.Name,
+                        document.DetectedLanguage.Value.Iso6391Name,
+                        document.DetectedLanguage.Value.ConfidenceScore,
+                        document.DetectedLanguage.Value.Script,
+                        default)
+                    : null;
```

Comment:
Saw this section of code being used multiple places, should we create an utility method which can be reused?